cardano_node_tests/utils/dbsync_utils.py

```diff
@@ -848,6 +849,168 @@ def check_param_proposal(protocol_params: dict) -> tp.Optional[dbsync_queries.Pa
+        "coins_per_utxo_size": pp.get("coinsPerUTxOByte"),
+        "min_fee_ref_script_cost_per_byte": pp.get("minFeeRefScriptCostPerByte"),
+        "price_mem": price_mem,
+        "price_step": price_steps,
+        # Technical proposals group
+        "influence": _evaluate_pparam(pp.get("poolPledgeInfluence")),
+        "max_epoch": pp.get("poolRetireMaxEpoch"),
+        "optimal_pool_count": pp.get("stakePoolTargetNum"),
+        "collateral_percent": pp.get("collateralPercentage"),
+        # Governance proposal group
+        # - DReps
+        "dvt_committee_no_confidence": _evaluate_pparam(dvt.get("committeeNoConfidence")),
+        "dvt_committee_normal": _evaluate_pparam(dvt.get("committeeNormal")),
+        "dvt_hard_fork_initiation": _evaluate_pparam(dvt.get("hardForkInitiation")),
+        "dvt_motion_no_confidence": _evaluate_pparam(dvt.get("motionNoConfidence")),
+        "dvt_p_p_economic_group": _evaluate_pparam(dvt.get("ppEconomicGroup")),
+        "dvt_p_p_gov_group": _evaluate_pparam(dvt.get("ppGovGroup")),
+        "dvt_p_p_network_group": _evaluate_pparam(dvt.get("ppNetworkGroup")),
+        "dvt_p_p_technical_group": _evaluate_pparam(dvt.get("ppTechnicalGroup")),
+        "dvt_treasury_withdrawal": _evaluate_pparam(dvt.get("treasuryWithdrawal")),
+        "dvt_update_to_constitution": _evaluate_pparam(dvt.get("updateToConstitution")),
+        # - Pools
+        "pvt_committee_no_confidence": _evaluate_pparam(pvt.get("committeeNoConfidence")),
+        "pvt_committee_normal": _evaluate_pparam(pvt.get("committeeNormal")),
+        "pvt_hard_fork_initiation": _evaluate_pparam(pvt.get("hardForkInitiation")),
+        "pvt_motion_no_confidence": _evaluate_pparam(pvt.get("motionNoConfidence")),
+        "pvtpp_security_group": _evaluate_pparam(pvt.get("ppSecurityGroup")),
+        # General
+        "gov_action_lifetime": pp.get("govActionLifetime"),
+        "gov_action_deposit": pp.get("govActionDeposit"),
+        "drep_deposit": pp.get("dRepDeposit"),
+        "drep_activity": pp.get("dRepActivity"),
+        "committee_min_size": pp.get("committeeMinSize"),
+        "committee_max_term_length": pp.get("committeeMaxTermLength"),
+    }
+
+    return params_mapping
+
+
+def _check_param_proposal(
+    param_proposal_db: tp.Union[dbsync_queries.ParamProposalDBRow, dbsync_queries.EpochParamDBRow],
+    params_map: dict,
+) -> list:
+    """Check parameter proposal against db-sync."""
+    failures = []
+
+    for param_db, protocol_value in params_map.items():
+        if protocol_value:
+            db_value = getattr(param_proposal_db, param_db)
+            if db_value and (db_value != protocol_value):
+                failures.append(
+                    f"Param value for {param_db}: {db_value}. Expected: {protocol_value}"
+                )
+    return failures
+
+
+def check_conway_param_update_proposal(
+    param_proposal_ledger: dict,
+) -> tp.Optional[dbsync_queries.ParamProposalDBRow]:
```

Comment:
For functions called directly from tests, we need to check that we have db-sync available.